engine/common/follower/engine.go

```diff
@@ -381,6 +386,36 @@ func (e *Engine) processBlockAndDescendants(ctx context.Context, proposal *messa
 
 	log.Info().Msg("processing block proposal")
 
+	// retrieve the parent
+	parent, err := e.headers.ByBlockID(header.ParentID)
+	if err != nil {
+		return fmt.Errorf("could not retrieve proposal parent: %w", err)
+	}
+
+	hotstuffProposal := model.ProposalFromFlow(header, parent.View)
+	err = e.validator.ValidateProposal(hotstuffProposal)
+	if err != nil {
+		if model.IsInvalidBlockError(err) {
+			// TODO potential slashing
+			log.Err(err).Msgf("received invalid block proposal (potential slashing evidence)")
+			return nil
+		}
+		if errors.Is(err, model.ErrViewForUnknownEpoch) {
+			// We have received a proposal, but we don't know the epoch its view is within.
+			// We know:
+			//  - the parent of this block is valid and inserted (ie. we knew the epoch for it)
+			//  - if we then see this for the child, one of two things must have happened:
+			//    1. the proposer malicious created the block for a view very far in the future (it's invalid)
+			//      -> in this case we can disregard the block
+			//    2. no blocks have been finalized the epoch commitment deadline, and the epoch end
+			//       (breaking a critical assumption - see EpochCommitSafetyThreshold in protocol.Params for details)
+			//      -> in this case, the network has encountered a critical failure
+			//  - we assume in general that Case 2 will not happen, therefore we can discard this proposal
+			log.Err(err).Msg("unable to validate proposal with view from unknown epoch")
```

Comment:
> therefore we can discard this proposal

To disregard the proposal, wouldn't we need to return here? Otherwise, we trigger an exception in [line 416](https://github.com/onflow/flow-go/blob/b4a7dbb4a6b7cf09a481f7eaa49c5bd8057f2427/engine/common/follower/engine.go#L416)

```suggestion
			log.Err(err).Msg("unable to validate proposal with view from unknown epoch")
			return nil
```

Reply:
Hmm, we do return already 🤔

Reply:
fixed in https://github.com/onflow/flow-go/pull/3408